src/pages/Intents/intentsDatastoreQuery.js

```diff
@@ -0,0 +1,76 @@
+import { useDataMutation, useDataQuery } from '@dhis2/app-runtime'
+import isEmpty from 'lodash/isEmpty'
+import { useEffect, useState } from 'react'
+import { CUSTOM_INTENTS, NAMESPACE } from '../../constants/data-store'
+import { createKeyCustomIntent, updateSharingMutation } from '../../modules'
+
+/**
+ * update data store
+ * key: Custom intents
+ * */
+
+export const saveCustomIntentsMutation = {
+    resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}`,
+    type: 'update',
+    data: ({ settings }) => ({
+        ...settings,
+        lastUpdated: new Date().toJSON(),
+    }),
+}
+
+/**
+ * Query to get Customs Intents from Data store
+ * key: Custom Intents
+ * */
+
+export const getCustomIntentsQuery = {
+    customIntents: {
+        resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}`,
+    },
+}
+
+const getMetadataQuery = {
+    customIntents: {
+        resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}/metaData`,
+    },
+}
+
+export const useReadCustomIntentsDataStore = () => {
+    const { data, loading: loadingNamespace } = useDataQuery(
+        getCustomIntentsQuery
+    )
+    const [mutate] = useDataMutation(createKeyCustomIntent)
```

Comment:
I feel like renaming the `mutate` and `refetch` here would be helpful for ease of readability, e.g. `mutate: createKeyCustomIntent`, refetch: setSharing`)